src/poetry/utils/helpers.py

```diff
@@ -33,15 +31,6 @@ def module_name(name: str) -> str:
     return canonicalize_name(name).replace(".", "_").replace("-", "_")
 
 
-@contextmanager
-def temporary_directory(*args: Any, **kwargs: Any) -> Iterator[str]:
-    name = tempfile.mkdtemp(*args, **kwargs)
-
-    yield name
-
-    remove_directory(name, force=True)
```

Comment:
Note the force implementation in `remove_directory`. The reason why it exists is twofolds.

1. Git directories can be read only causing issues on windows.  #1243
2. Certain sdists can have similar issues. #520.

Will probably need to verify that the temfile implementation handles that well.

Reply:
I'd have much more confidence in the language implementation of `TemporaryDirectory()` than in a home-grown version when it comes to dealing with edge cases!

And indeed it looks to do all that remove_directory() does and more: https://github.com/python/cpython/blob/e91dee87edcf6dee5dd78053004d76e5f05456d4/Lib/tempfile.py#L808-L836

Reply:
I agree the system implementation is preferred. However, I'd want to make sure we are not regressing here somehow.

Reply:
Well, there's a testcase over in poetry-core for this.  So the pipeline from https://github.com/python-poetry/poetry-core/pull/337 should tell us.

Reply:
Colour me surprised, recent releases behave as expected but Windows and python 3.7 do indeed go wrong.

So I guess this will have to wait until such time as python 3.7 is out of support,

Reply:
https://bugs.python.org/issue26660

Reply:
We should add some comments around it and possibly move it into `_compat` to signal it's intent .